Sources/core/objects/Projectiles.cpp

```diff
@@ -0,0 +1,254 @@
+#include "Projectiles.h"
+#include "../Helpers.h"
+
+
+Projectile::Projectile(size_t id) : IMovable(id)
+{
+    SetPassable(D_ALL, Passable::AIR);
+    tickSpeed = 0.25;
+    damage = 0;
+    SetSprite("icons/projectiles.dmi");
+    v_level = 5;
+}
+void Projectile::Process()
+{
+    if(set_target_ == 0)
+    {
+       return;
+    }
+    int z = current_step_;
+    Dir step = VDirToDir(movement_[z]);
+
+   if(!TryMove(step))
```

Comment:
You may have added some `VDir` constants, but it does not add functionality to other functions. Your segmentation fault probably caused because you use diagonal dirs (you have not updated `DirToVDir`  array for example). Noone in the current Griefly world does not expect to have any interaction with something that has a diagonal direction, and it may cause unknown consequences.
Do not use diagonal dirs, they add to much new functionality, and they are not needed for the current state of the game.